src/trix/models/composition.coffee

```diff
@@ -217,11 +227,11 @@ class Trix.Composition extends Trix.BasicObject
       @removeCurrentAttribute(attributeName)
 
   canSetCurrentAttribute: (attributeName) ->
-    switch attributeName
-      when "href"
-        not @selectionContainsAttachmentWithAttribute(attributeName)
-      else
-        true
+    return not @selectionContainsAttachmentWithAttribute(attributeName) if attributeName is "href"
+    return true if attributeName is @getBlock()?.getLastAttribute()
+    if @getBlock()?.hasAttributes() and @getBlock()?.getConfig("terminal")
+      return false
```

Comment:
I think the `switch` expression was easier to read:

```coffee
switch attributeName
  when "href"
    not @selectionContainsAttachmentWithAttribute(attributeName)
  when getBlock()?.getLastAttribute()
    true
  else
    not (@getBlock()?.hasAttributes() and @getBlock()?.getConfig("terminal"))
```

Also, let’s extract a method for the terminal condition.